src/pages/index.js

```diff
@@ -1,60 +1,123 @@
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
-import {initialCards, selectors, validationSettings, profileEditForm, addCardForm, nameInput, aboutInput} from "../utils/constants.js";
+import { selectors, validationSettings, profileEditForm, addCardForm, editAvatarForm, nameInput, aboutInput } from "../utils/constants.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
+
+// Initialize api
+const api = new Api({baseUrl: "https://around-api.en.tripleten-services.com/v1", authToken: "9ef0145a-4550-4d8d-9dc2-252c6d573e29"});
+
+
+// create delete card popup
+const deleteCardPopup = new PopupWithConfirmation({
+  popupSelector: selectors.deleteCardPopup,
+  handleConfirm: (cardId) => {
+    api.deleteCard(cardId);
+  }
+});
 
 // Card renderer function
 const renderCard = (data) => {
   const cardElement = new Card(
     {
-      data,
+      data: data,
       handleImageClick: (imgData) => {
         cardPreviewPopup.open(imgData);
-      }
+      },
+      handleDelete: (card) => {
+        deleteCardPopup.open(card);
+        deleteCardPopup.setSubmitAction((card) => {
+          api.deleteCard(card.getId());
+          card.removeCard();
+        });
+      },
+      confirmPopup: deleteCardPopup,
+      api: api
     },
     selectors.cardTemplate
   );
   cardSection.addItem(cardElement.generateCard());
 }
 
 
+
+// Get initial user information and cards
+api.getUserInfo()
+  .then(res => {
+    document.querySelector(".profile__title").textContent = res.name;
+    document.querySelector(".profile__description").textContent = res.about;
+    document.querySelector(".profile__image").src = res.avatar;
+  });
+
+api.getInitialCards()
+  .then(res => {
+    res.reverse().forEach((card) => {
+      renderCard(card);
+    });
+  });
+
+
+
 // Create card and card image popup instances
 export const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
 export const cardSection = new Section({ renderer: renderCard }, selectors.cardList);
 
 
+
 // Create form and user info instances
 const userInfo = new UserInfo({nameSelector: selectors.nameElement, aboutMeSelector: selectors.descriptionElement});
 
 const editProfilePopup = new PopupWithForm(selectors.editProfilePopup, (data) => {
   userInfo.setUserInfo(data);
+  return api.updateProfileInfo();
 });
 
 const addCardPopup = new PopupWithForm(selectors.addCardPopup, (data) => {
-  renderCard(data);
+  return api.addCard(data)
+    .then((res) => {
+      renderCard(res);
+    });
 });
 
+const editAvatarPopup = new PopupWithForm(selectors.editAvatarPopup, (data) => {
+  return api.updateAvatar(data)
+    .then((res) => {
+      document.querySelector(".profile__image").src = res.avatar;
+    })
+});
+
+
+
 
 // Form validation
 const editFormValidator = new FormValidator(validationSettings, profileEditForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```